atomic-internal.h

```diff
@@ -0,0 +1,80 @@
+#ifndef ATOMIC_INTERNAL_H_INCLUDED_
+#define ATOMIC_INTERNAL_H_INCLUDED_
+
+/*
+ * Defines atomic macros/functions for intptr_t sized variables.
+ * _load, _store, _exchange, _compare_exchange_strong are included.
+ * Use atomic_init_ptr to declare variables.
+ *
+ * Microsoft explicitly excludes stdatomic.h from the list of supported
+ * C11 features on Windows. Implement equivalent functions using the
+ * Interlocked API.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdbool.h>
+
+#if defined(_MSC_VER)
+#include <intrin.h>
+#else
+#include <stdatomic.h>
+#endif
+
+#include "event2/event-config.h"
+
+#if defined(_MSC_VER)
+
+#if EVENT__SIZEOF_UINTPTR_T == 8
+#define _Interlocked(f) _Interlocked##f##64
+#elif EVENT__SIZEOF_UINTPTR_T == 4
+#define _Interlocked(f) _Interlocked##f
+#else
+#error "invalid value for EVENT__SIZEOF_UINTPTR_T"
+#endif
+
+static inline bool atomic_compare_exchange_strong(intptr_t volatile *p, intptr_t *expected, intptr_t desired)
+{
+	intptr_t old = _Interlocked(CompareExchange)(p, desired, *expected);
+	if (old == *expected)
+		return true;
+
+	*expected = old;
+	return false;
+}
+
+static inline void *atomic_load(intptr_t volatile *p)
```

Comment:
if i'm not mistaken, any ordering lower than `seq_cst` will not suffice to prevent races from co-dependent access to `_debug_mode` and `_too_late`.
there might be a better solution but, afaics, this is the best approach that avoids having to protect both variables using a mutex (which would be costlier).